pytorch_toolkit/ote/ote/modules/arg_converters/mmdetection.py

```diff
@@ -63,8 +64,83 @@ class MMDetectionArgsConverter(BaseArgConverter):
         'test_data_roots': 'data.test.img_prefix',
     }
 
+
+def load_classes_from_snapshot(snapshot):
+    import torch
+    return torch.load(snapshot)['meta'].get('CLASSES', [])
+
+def classes_list_to_update_config_dict(cfg, classes):
+    if isinstance(cfg, str):
+        from mmcv import Config
+        cfg = Config.fromfile(cfg)
+    num_classes = len(classes)
+    classes = '[' + ','.join(f'"{x}"' for x in classes) + ']'
+    update_config_dict = {
+        'data.train.dataset.classes': classes,
+        'data.val.classes': classes,
+        'data.test.classes': classes,
+        'model.bbox_head.num_classes': num_classes
+    }
+    if hasattr(cfg.model, 'roi_head'):
+        if 'mask_head' in cfg.model.roi_head.keys():
+            update_config_dict['model.roi_head.mask_head.num_classes'] = num_classes
+    return update_config_dict
+
+
+@ARG_CONVERTERS.register_module()
+class MMDetectionCustomClassesArgsConverter(MMDetectionArgsConverter):
+
+    @staticmethod
+    def _get_classes_from_annotation(annotation_file):
+        with open(annotation_file) as read_file:
+            categories = sorted(json.load(read_file)['categories'], key=lambda x: x['id'])
+        classes_from_annotation = [category_dict['name'] for category_dict in categories]
+        return classes_from_annotation
+
     def _get_extra_train_args(self, args):
```

Comment:
2 @Ilya-Krylov 
Ilya, unfortunately I have to say that this function `_get_extra_train_args` is not sufficient for compression.
At the moment the base arg converter uses the function `_get_extra_train_args` both for train and compression -- see the method `convert_compress_args` in `arg_converters/base.py`
As you can see there is `TODO(LeonidBeynenson): think on _get_extra_compress_args` over there.
IMHO now it should be implemented. 
(Unfortunately for me, since it suspends my last PR that I hoped to merge soon)
Could you, implement it in `BaseArgConverter` (by default the same as `_get_extra_train_args` and `_get_extra_test_args`).
I think for `compress.py` the function `_get_extra_compress_args` should return `_get_extra_test_args`
It will also receive class info from `.extra_params` , but IMHO it is OK.

Reply:
@LeonidBeynenson As far as I understand, not it also works as well, let's avoid complicating this PR.

Reply:
It works, but it won't raise `RuntimeError` if we pass another numer of classes through args

Reply:
Fixed!